deps/rabbitmq_mqtt/src/rabbit_mqtt_packet.erl

```diff
@@ -131,10 +131,10 @@ parse_connect(Bin, Fixed, Length) ->
                             wrap(Fixed,
                                  #mqtt_packet_connect{
                                     proto_ver   = ProtoVersion,
-                                    will_retain = bool(WillRetain),
+                                    will_retain = int_to_bool(WillRetain),
                                     will_qos    = WillQos,
-                                    will_flag   = bool(WillFlag),
-                                    clean_sess  = bool(CleanSession),
+                                    will_flag   = int_to_bool(WillFlag),
+                                    clean_sess  = int_to_bool(CleanSession),
```

Comment:
Is there any benefit of having `CleanSession`, `WillFlag `, and `WillRetain` as boolean? If you use the integer value directly, maybe you could avoid converting int <=> bool.

Reply:
Technically, one could use an integer instead of boolean (atom) throughout the code base. However, this conversion is extremely cheap. The benefit of having booleans (atoms) is much more readable code as for example in:
https://github.com/rabbitmq/rabbitmq-server/blob/41f98ccbefddaddb6374d69a0c8eb8af746d69ee/deps/rabbitmq_mqtt/src/rabbit_mqtt_processor.erl#L430
https://github.com/rabbitmq/rabbitmq-server/blob/41f98ccbefddaddb6374d69a0c8eb8af746d69ee/deps/rabbitmq_mqtt/src/rabbit_mqtt_processor.erl#L1241-L1246

and clearer log output as for example in:
https://github.com/rabbitmq/rabbitmq-server/blob/41f98ccbefddaddb6374d69a0c8eb8af746d69ee/deps/rabbitmq_mqtt/src/rabbit_mqtt_processor.erl#L123-L125
printing `false` instead of `0`.